### PR TITLE
Add Dokku app creation

### LINKED ISSUE #42: Create Dokku app alongside DB record

### PR DESCRIPTION
**Pull request by cjdenio**
Closes #42

### CODE REVIEW

**Comment by anirudhb** [comment]
Can we split this into smaller PRs? (Or get this merged soon) A lot of fundamental structure changes are introduced here which are necessary for other unrelated features.

**Comment by cjdenio** [comment]
Sounds good; I sometimes get a bit too wild with PRs lol

File: pkg/api/apps/create.go [hunk 1]
```diff
@@ -17,19 +38,33 @@ func handlePOSTApp(c *gin.Context) {
...
+
 	// Check that the given team ID exists
 	result := db.DB.First(&db.Team{}, "id = ?", json.TeamID)
 	if result.Error != nil {
 		c.JSON(400, gin.H{"status": "error", "message": "Invalid team ID"})
 		return
 	}
 
+	// Create Dokku app
+	_, err = dokku_conn.RunCommand(context.Background(), []string{"apps:create", json.ShortName})
```
Comment (anirudhb) [inline pkg/api/apps/create.go]: use the request context here

**Review by anirudhb — CHANGES_REQUESTED** [review]
lgtm, nits

**Review by anirudhb — APPROVED** [review]
lgtm

**Pull request merged**